### ISSUE TITLE
Debugging using source code of Django isn't working

### ISSUE THREAD

**Issue by Himanshu-Balasamanta** [microsoft/debugpy#872]
Issue Type: <b>Bug</b>

# Behaviour
## Expected vs. Actual

Throws ImportError in manage.py saying django isn't present, check PYTHONPATH and virtualenv 
XXX

## Steps to reproduce:

1. Clone through src code from github
2. pip install -e django
3. Use python debugger for django to run your application

# Diagnostic data

-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.10
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): Venv
-   Value of the `python.languageServer` setting: Pylance

<details>

<summary>User Settings</summary>

<p>

```

defaultLS: {"defaultLSType":"Pylance"}

downloadLanguageServer: true

envFile: "<placeholder>"

venvPath: "<placeholder>"

venvFolders: "<placeholder>"

condaPath: "<placeholder>"

pipenvPath: "<placeholder>"

poetryPath: "<placeholder>"

languageServer: "Pylance"

linting
• enabled: true
• cwd: "<placeholder>"
• Flake8Args: "<placeholder>"
• flake8Enabled: false
• flake8Path: "<placeholder>"
• lintOnSave: true
• banditArgs: "<placeholder>"
• banditEnabled: false
• banditPath: "<placeholder>"
• mypyArgs: "<placeholder>"
• mypyEnabled: false
• mypyPath: "<placeholder>"
• pycodestyleArgs: "<placeholder>"
• pycodestyleEnabled: false
• pycodestylePath: "<placeholder>"
• prospectorArgs: "<placeholder>"
• prospectorEnabled: false
• prospectorPath: "<placeholder>"
• pydocstyleArgs: "<placeholder>"
• pydocstyleEnabled: false
• pydocstylePath: "<placeholder>"
• pylamaArgs: "<placeholder>"
• pylamaEnabled: false
• pylamaPath: "<placeholder>"
• pylintArgs: "<placeholder>"
• pylintPath: "<placeholder>"

sortImports
• args: "<placeholder>"
• path: "<placeholder>"

formatting
• autopep8Args: "<placeholder>"
• autopep8Path: "<placeholder>"
• provider: "autopep8"
• blackArgs: "<placeholder>"
• blackPath: "<placeholder>"
• yapfArgs: "<placeholder>"
• yapfPath: "<placeholder>"

testing
• cwd: "<placeholder>"
• debugPort: 3000
• nosetestArgs: "<placeholder>"
• nosetestsEnabled: undefined
• nosetestPath: "<placeholder>"
• promptToConfigure: true
• pytestArgs: "<placeholder>"
• pytestEnabled: false
• pytestPath: "<placeholder>"
• unittestArgs: "<placeholder>"
• unittestEnabled: false
• autoTestDiscoverOnSaveEnabled: true

terminal
• activateEnvironment: true
• executeInFileDir: "<placeholder>"
• launchArgs: "<placeholder>"

experiments
• enabled: true
• optInto: []
• optOutFrom: []

insidersChannel: "off"

tensorBoard
• logDirectory: "<placeholder>"

```

</p>
</details>

Extension version: 2022.2.1924087327
VS Code version: Code 1.65.2 (c722ca6c7eed3d7987c0d5c3df5c45f6b15e77d1, 2022-03-10T15:36:26.048Z)
OS version: Linux x64 5.4.0-94-generic snap
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD A6-7310 APU with AMD Radeon R4 Graphics (4 x 1959)|
|GPU Status|2d_canvas: unavailable_software<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: disabled_off<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software|
|Load (avg)|0, 1, 1|
|Memory (System)|6.75GB (0.28GB free)|
|Process Argv|--no-sandbox --force-user-env --unity-launch --crash-reporter-id 09a85e9a-1856-433f-a15b-cb74adf93ae3|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|x11|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30443607
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
vsclayoutctrt:30451275
dsvsc009:30452663
pythonvspyt640:30450904
vscscmwlcmt:30438805
pynewfile477:30450038

```

</details>

**Comment by fabioz** [comment]
Well, the errors seem correct as you're having an import error because django couldn't be imported (which points to a wrong `PYTHONPATH` configuration).

Can you print your `sys.path` (with the code below):

```
import sys
print("\n".join(sorted(sys.path)))
```

and point to where `django` itself resides in your machine?

**Comment by Himanshu-Balasamanta** [comment]
I had taken that obvious step. Didn't work. Normally running python3 manage.py works fine with the src code, but using vscode debugger throws that error

**Comment by karthiknadig** [comment]
@Himanshu-Balasamanta Can you scroll up to where the debugger command is seen? Do you see if the environment is activated before running the command to launch your django app?

**Comment by int19h** [comment]
`PYTHONPATH` is supposed to be pointing to the folder in which packages are to be found, rather than the package itself - but it looks like you're pointing it at the django package? Or do you have `${workspaceFolder}/django/django`? If not, you should remove the last segment of your `PYTHONPATH` entry.

**Comment by Himanshu-Balasamanta** [comment]
Thank you so much @karthiknadig and @int19h , but I had verified both your points already before opening this issue.

**Comment by Himanshu-Balasamanta** [comment]
I got the solution about an hour ago.
I saw VSCode by default uses python from the site-packages from the virtual env, but anything working using python3 on terminal would be installed into global site-packages.
The command run by VSCode debugger is
`
cd /home/new_developer/DEVELOPER-TOOLS/Django ; /usr/bin/env /home/new_developer/DEVELOPER-TOOLS/Django/.djangodev/bin/python /home/new_developer/.vscode/extensions/ms-python.python-2022.2.1924087327/pythonFiles/lib/python/debugpy/launcher 45101 -- /home/new_developer/DEVELOPER-TOOLS/Django/mysite/manage.py runserver 
`

In the pyvenv.cfg in my virtual environment, I had to set `include-system-site-packages = true` . This by default was false.
As a result python3 commands which used global site-packages were now available to my virtual environment.
So now VSCode when debugging, uses site-packages that were globally installed.

It is now working.
